src/v/serde/serde.h

```diff
@@ -132,6 +132,14 @@ void write(iobuf& out, T t) {
     } else if constexpr (std::is_scalar_v<Type> && !std::is_enum_v<Type>) {
         if constexpr (sizeof(Type) == 1) {
             out.append(reinterpret_cast<char const*>(&t), sizeof(t));
+        } else if constexpr (std::is_same_v<float, Type>) {
+            auto const le_t = htole32(t);
+            static_assert(sizeof(le_t) == sizeof(Type));
+            out.append(reinterpret_cast<char const*>(&le_t), sizeof(le_t));
+        } else if constexpr (std::is_same_v<double, Type>) {
+            auto const le_t = htole64(t);
+            static_assert(sizeof(le_t) == sizeof(Type));
+            out.append(reinterpret_cast<char const*>(&le_t), sizeof(le_t));
```

Comment:
Having a look over at https://github.com/ceph/ceph/blob/master/src/include/encoding.h#L143 it seems like we may also want to assert that the expected floating point standard is being used. There are also some extra details in there that may be of interest.